File: backend/src/main/java/com/petcare/model/GrowthTracking.java
package com.petcare.model;

import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDate;

@Entity
@Table(name = "growth_tracking")
@Data
public class GrowthTracking {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(name = "pet_id", nullable = false)
    private Long petId;
    
    @Column(name = "tracking_date", nullable = false)
    private LocalDate trackingDate;
    
    @Column(nullable = false)
    private Double weight;
    
    @Column(nullable = false)
    private Double height;
    
    @Column(name = "age_months")
    private Integer ageMonths;
    
    @Column(columnDefinition = "TEXT")
    private String notes;
}

File: backend-python/app/models/growth_tracking.py
from sqlalchemy import Column, Integer, Float, Date, Text
from app.database import Base

class GrowthTracking(Base):
    __tablename__ = "growth_tracking"
    
    id = Column(Integer, primary_key=True, index=True)
    pet_id = Column(Integer, nullable=False, index=True)
    tracking_date = Column(Date, nullable=False)
    weight = Column(Float, nullable=False)
    height = Column(Float, nullable=False)
    age_months = Column(Integer)
    notes = Column(Text)
